src/mlpack/bindings/go/print_doc_functions_impl.hpp

```diff
@@ -0,0 +1,328 @@
+
+  return result;
+}
+
+
+/**
+ * Given a name of a binding and a variable number of arguments (and their
+ * contents), print the corresponding function call.
+ */
+template<typename... Args>
+std::string ProgramCall(const std::string& programName, Args... args)
+{
+  std::string result = "";
+  std::string goProgramName = programName;
+  goProgramName[0] = std::toupper(goProgramName[0]);
+
+  // Initialize the method parameter structure
+  std::ostringstream oss;
+  oss << ">>> param := Initialize" << goProgramName << "()\n";
+  result = oss.str();
+  oss.str(""); // Reset it.
+
+  // Now process each optional parameters.
+  oss << PrintOptionalInputs(args...);
+  std::string param = oss.str();
+  result = result + util::HyphenateString(param, 0);
+  oss.str(""); // Reset it.
+
+  // Now process each output parameters.
+  oss << PrintOutputOptions(args...);
+  std::string output = oss.str();
+  result = result + ">>> " + util::HyphenateString(output, 0);
+  oss.str(""); // Reset it.
+
+  oss << " := " << goProgramName << "(";
+  result = result + oss.str();
+  oss.str(""); // Reset it.
+
+  // Now process each input required parameters.
+  oss << PrintInputOptions(args...);
+  std::string input = oss.str();
+  if (input != "")
+    result = result + input + ", ";
+  oss.str(""); // Reset it.
+  if (param != "")
+   result = result + "param";
+
+  result = result + ")";
+  return result;
+}
+
+/**
+ * Given the name of a model, print it.  Here we do not need to modify anything.
+ */
+inline std::string PrintModel(const std::string& modelName)
+{
+  std::string goModelName = modelName;
+  const size_t loc = goModelName.find("<>");
+  if (loc != std::string::npos)
+  {
+    // Convert it from "<>".
+    goModelName.replace(loc, 2, "");
+  }
+  return goModelName;
+}
+
+/**
+ * Given the name of a matrix, print it.  Here we do not need to modify
+ * anything.
+ */
+inline std::string PrintDataset(const std::string& datasetName)
+{
+  return "" + datasetName + "";
+}
+
+/**
+ * Given the name of a binding, print its invocation.
+ */
+inline std::string ProgramCall(const std::string& programName)
+{
+  std::string goProgramName = programName;
+  goProgramName[0] = std::toupper(goProgramName[0]);
+  return goProgramName + "(";
+}
+
+/**
+ * Print any closing call to a program.
+ */
+inline std::string ProgramCallClose()
+{
+  return ")";
+}
+
+/**
+ * Given the parameter name, determine what it would actually be when passed to
+ * the command line.
+ */
+inline std::string ParamString(const std::string& paramName)
+{
+  // For a Go binding we don't need to know the type.
+
+  return "'" + paramName + "'";
+}
+
+/**
+ * Given the parameter name and an argument, return what should be written as
+ * documentation when referencing that argument.
+ */
+template<typename T>
+inline std::string ParamString(const std::string& paramName, const T& value)
+{
+  std::ostringstream oss;
+    oss << paramName << "=" << value;
+  return oss.str();
```

Comment:
I think that this function will actually never be called, judging by all the calls to `PRINT_PARAM_STRING()` that I find in the `src/mlpack/methods/` folders.  So maybe it can be removed?